Blizzard.IO.RabbitMQ/Rpc/NetqRabbitRpcClient.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Blizzard.IO.Core.Rpc;
+using Blizzard.IO.RabbitMQ.Entities.Rpc;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace Blizzard.IO.RabbitMQ.Rpc
+{
+    public class NetqRabbitRpcClient : IRpcClient
+    {
+        private readonly INetqRabbitRpcConnection<Func<Type, object>> _netqRpcRabbitConnection;
+        private readonly ILogger _logger;
+
+        public NetqRabbitRpcClient(INetqRabbitRpcConnection<Func<Type, object>> netqRpcRabbitConnection, ILogger logger)
+        {
+            _logger = logger;
+            _netqRpcRabbitConnection = netqRpcRabbitConnection;
+        }
+
+        public TRespond Request<TRequest, TRespond>(TRequest request)
+        where TRequest : class
+        where TRespond : class
+        {
+            if (_netqRpcRabbitConnection.RpcMessageType == RpcMessageType.Abstract)
+            {
+                _logger.LogDebug($"Requesting an abstract type request of type: {typeof(TRequest)} and expecting to receive a respond of type: {typeof(TRespond)}");
+                return _netqRpcRabbitConnection.Bus.Request<TRequest, TRespond>(request);
+            }
+            else if (_netqRpcRabbitConnection.RpcMessageType == RpcMessageType.Concrete)
+            {
+                _logger.LogDebug($"Requesting a concrete type request of type: {typeof(TRequest)} and expecting to receive a respond of type: {typeof(TRespond)}");
+                var func = _netqRpcRabbitConnection.Bus.Request<TRequest, Func<Type, object>>(request);
+                return (TRespond)func(typeof(TRespond));
+            }
+
+            throw new InvalidOperationException($"Cannot handle request {nameof(_netqRpcRabbitConnection)} rpc type is invalid, " +
```

Comment:
The user must choose a strategy, and if he doesn't we set the default one to `Concrete`, there's no good reason for the code to throw this exception. On the contrary, if the chosen serialization strategy fails an exception should be thrown

Reply:
I dont want to deceive the user into thinking his strategy was used when in reality I used this strategy